File: frontend/src/types/prompt.ts
export type Prompt = {
  id: number;
  user_id: number;
  user_update_id: number | null;
  type: string;
  short_description: string;
  text: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  creator_username: string | null;
};

export type PromptCreateRequest = {
  type: string;
  short_description: string;
  text: string;
};

export type PromptUpdateRequest = {
  type?: string;
  short_description?: string;
  text?: string;
};

export type PromptListResponse = {
  items: Prompt[];
  total: number;
  offset: number;
  limit: number;
};

File: backend/app/schemas/prompt.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PromptBase(BaseModel):
    type: str = Field(min_length=1, max_length=100)
    short_description: str = Field(min_length=1, max_length=200)
    text: str = Field(min_length=1)


class PromptCreate(PromptBase):
    pass


class PromptUpdate(BaseModel):
    type: str | None = Field(default=None, min_length=1, max_length=100)
    short_description: str | None = Field(default=None, min_length=1, max_length=200)
    text: str | None = Field(default=None, min_length=1)


class PromptRead(BaseModel):
    id: int
    user_id: int
    user_update_id: int | None
    type: str
    short_description: str
    text: str
    is_active: bool
    created_at: datetime
    updated_at: datetime
    creator_username: str | None = None

    model_config = ConfigDict(from_attributes=True)


class PromptListResponse(BaseModel):
    items: list[PromptRead]
    total: int
    offset: int
    limit: int